src/lib.rs

```diff
@@ -142,15 +142,7 @@ fn decode_ac(value: usize, maximum_value: f64) -> [f64; 3] {
 }
 
 fn sign_pow(value: f64, exp: f64) -> f64 {
-    get_sign(value) * f64::powf(f64::abs(value), exp)
-}
-
-fn get_sign(n: f64) -> f64 {
-    if n < 0f64 {
-        -1f64
-    } else {
-        1f64
-    }
+    value.abs().powf(exp).copysign(value)
 }
```

Comment:
While this isn't related to preallocation, this bespoke sign copy function was easily standardized, so I just included it in this PR.

Reply:
Nice! I was wondering if such a function existed. Makes it nicer to read as well!